bin/reth/src/args/rpc_server_args.rs

```diff
@@ -99,12 +105,60 @@ impl RpcServerArgs {
             client,
             pool,
             network,
-            TransportRpcModuleConfig::default()
-                .with_http(vec![RethRpcModule::Admin, RethRpcModule::Eth]),
-            RpcServerConfig::default().with_http(Default::default()),
+            self.transport_rpc_module_config(),
+            self.rpc_server_config(),
         )
         .await
     }
+
+    fn transport_rpc_module_config(&self) -> TransportRpcModuleConfig {
+        let mut config = TransportRpcModuleConfig::default();
+        let rpc_modules =
+            RpcModuleSelection::Selection(vec![RethRpcModule::Admin, RethRpcModule::Eth]);
+        if self.http {
+            config = config.with_http(self.http_api.as_ref().unwrap_or(&rpc_modules).clone());
+        }
+
+        if self.ws {
+            config = config.with_ws(self.ws_api.as_ref().unwrap_or(&rpc_modules).clone());
+        }
+
+        TransportRpcModuleConfig::from(config)
+    }
+
+    fn rpc_server_config(&self) -> RpcServerConfig {
+        let mut config = RpcServerConfig::default();
+
+        if self.http {
+            let socket_address = SocketAddr::new(
+                self.http_addr.unwrap_or(IpAddr::V4(Ipv4Addr::UNSPECIFIED)),
+                self.http_port.unwrap_or(DEFAULT_HTTP_RPC_PORT),
+            );
+            config = config.with_http_address(socket_address).with_http(ServerBuilder::new());
+        }
+
+        if self.ws {
+            let socket_address = SocketAddr::new(
+                self.ws_addr.unwrap_or(IpAddr::V4(Ipv4Addr::UNSPECIFIED)),
+                self.ws_port.unwrap_or(DEFAULT_HTTP_RPC_PORT),
+            );
+            config = config.with_ws_address(socket_address).with_http(ServerBuilder::new());
+        }
+
+        if !self.ipcdisable {
+            let ipc_builder = IpcServerBuilder::default();
+            config = config.with_ipc(ipc_builder).with_ipc_endpoint(
+                self.ipcpath.as_ref().unwrap_or(&DEFAULT_IPC_ENDPOINT.to_string()),
+            );
+        }
+
+        RpcServerConfig::from(config)
+    }
+
+    #[allow(dead_code)]
+    async fn rpc_server(&self) -> Result<RpcServer, RpcError> {
```

Comment:
@mattsse I am not sure why need a function return `RpcServer`, it is just delegate `build` function.

Reply:
i think remove given the server fn above?

Reply:
this is still useful as a convenient function.

Reply:
how about add back when using it?